src/extensions/checkProjectBilling.ts

```diff
@@ -0,0 +1,130 @@
+}
+
+/**
+ * Opens URL if applicable and stalls until user responds.
+ */
+async function openBillingAccount(projectId: string, url: string, open: boolean): Promise<boolean> {
+  if (open) {
+    try {
+      opn(url);
+    } catch (err) {
+      logger.debug("Unable to open billing URL: " + err.stack);
+    }
+  }
+
+  await prompt.promptOnce({
+    name: "continue",
+    type: "confirm",
+    message:
+      "Press enter when finished upgrading your project to continue setting up your extension.",
+    default: true,
+  });
+  return cloudbilling.checkBillingEnabled(projectId);
+}
+
+/**
+ * Question prompts user to select billing account for project.
+ */
+async function chooseBillingAccount(
+  projectId: string,
+  extensionName: string,
+  accounts: cloudbilling.BillingAccount[]
+): Promise<void> {
+  const choices = accounts.map((m) => m.displayName);
+  choices.push("Add new billing account");
+
+  const answer = await prompt.promptOnce({
+    name: "billing",
+    type: "list",
+    message: `The extension ${clc.underline(
+      extensionName
+    )} requires your project to be upgraded to the Blaze plan. You have access to the following billing accounts.
+Please select the one that you would like to associate with this project:`,
+    choices: choices,
+  });
+
+  let billingEnabled: boolean;
+  if (answer === "Add new billing account") {
+    const billingURL = `https://console.cloud.google.com/billing/linkedaccount?project=${projectId}`;
+    billingEnabled = await openBillingAccount(projectId, billingURL, true);
+  } else {
+    const billingAccount = accounts.find((a) => a.displayName == answer);
+    billingEnabled = await cloudbilling.setBillingAccount(projectId, billingAccount!.name);
+  }
+
+  return logBillingStatus(billingEnabled, projectId);
+}
+
+/**
+ * Directs user to set up billing account over the web and stalls until
+ * user responds.
+ */
+async function setUpBillingAccount(projectId: string, extensionName: string) {
+  const billingURL = `https://console.cloud.google.com/billing/linkedaccount?project=${projectId}`;
+
+  logger.info();
+  logger.info(
+    `The extension ${clc.bold(
+      extensionName
+    )} requires your project to be upgraded to the Blaze plan. Please visit the following link to add a billing account:`
+  );
+  logger.info();
+  logger.info(clc.bold.underline(billingURL));
+  logger.info();
+
+  const open = await prompt.promptOnce({
+    name: "open-url",
+    type: "confirm",
+    message: "Press enter to open the URL.",
+    default: true,
+  });
+  const billingEnabled = await openBillingAccount(projectId, billingURL, open);
+  return logBillingStatus(billingEnabled, projectId);
+}
+
+/**
+ * Checks whether billing is enabled on the given project.
+ * @param {string} projectId
+ * @returns {Promise<boolean>} True if billing is enabled
+ */
+export async function isBillingEnabled(projectId: string): Promise<boolean> {
+  return cloudbilling.checkBillingEnabled(projectId);
```

Comment:
why do we need a helper for this?

Reply:
Honestly, not sure of the original intent. I suspect it was to simplify the dependency graph (you'd only need to import this instead of cloudbilling directly). Having said that, I'm all for getting rid of this.